calico/felix/fetcd.py

```diff
@@ -334,6 +335,56 @@ def __init__(self, config, etcd_api, status_reporter, hosts_ipset):
         self.last_rate_log_time = monotonic_time()
         # Register for events when values change.
         self._register_paths()
+        self._usage_report_greenlet = gevent.Greenlet(
+            self._periodically_usage_report
+        )
+
+    def estimated_host_count(self):
+        """
+        Single operations to python's built-in datastructures are thread-safe.
```

Comment:
To add a bit more background: I think the important thing is that the docstring should say what the function does and (if unusual, such as being thread safe) what guarantees it provides.  How it does that usually belongs in a comment.

The docstring tells you as the user of the API what you can/should/should not rely on.  The comment helps a future maintainer understand how the code works and why it was written that way.

Here, I think the docstring should say that it's safe to call this method from another thread.  The comment should say something like `# Thread safe because calling len() on a dict is atomic in Python.`